File: app/src/main/java/me/lukeforit/spaceofaday/di/FragmentContributorModule.java
package me.lukeforit.spaceofaday.di;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import me.lukeforit.spaceofaday.ui.archive.ApodArchiveFragment;
import me.lukeforit.spaceofaday.ui.pod.ApodDetailsFragment;

@Module
public abstract class FragmentContributorModule {
    @ContributesAndroidInjector
    abstract ApodArchiveFragment contributeApodArchiveFragment();

    @ContributesAndroidInjector
    abstract ApodDetailsFragment contributeApodDetailsFragment();
}

File: app/src/main/java/me/lukeforit/spaceofaday/di/FragmentContributorModule.kt
package me.lukeforit.spaceofaday.di

import dagger.Module
import dagger.android.ContributesAndroidInjector
import me.lukeforit.spaceofaday.ui.archive.ApodArchiveFragment
import me.lukeforit.spaceofaday.ui.pod.ApodDetailsFragment

@Module
abstract class FragmentContributorModule {
    @ContributesAndroidInjector
    internal abstract fun contributeApodArchiveFragment(): ApodArchiveFragment

    @ContributesAndroidInjector
    internal abstract fun contributeApodDetailsFragment(): ApodDetailsFragment
}
